src/main/java/com/techsophy/tsf/account/controller/impl/InternalControllerImpl.java

```diff
@@ -4,28 +4,52 @@
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.techsophy.tsf.account.config.GlobalMessageSource;
 import com.techsophy.tsf.account.controller.InternalController;
-import com.techsophy.tsf.account.dto.InternalUserFormDataSchema;
+import com.techsophy.tsf.account.dto.UserFormDataSchema;
+import com.techsophy.tsf.account.exception.BadRequestException;
+import com.techsophy.tsf.account.exception.RunTimeException;
+import com.techsophy.tsf.account.exception.UnAuthorizedException;
 import com.techsophy.tsf.account.model.ApiResponse;
-import com.techsophy.tsf.account.service.InternalService;
-import com.techsophy.tsf.account.utils.TokenUtils;
-import com.techsophy.tsf.account.utils.UserDetails;
+import com.techsophy.tsf.account.service.impl.UserFormDataServiceImpl;
+import com.techsophy.tsf.account.utils.Rsa4096;
 import lombok.AllArgsConstructor;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpHeaders;
 import org.springframework.web.bind.annotation.RestController;
 
+import javax.validation.ConstraintViolationException;
+
 import static com.techsophy.tsf.account.constants.AccountConstants.SAVE_FORM_SUCCESS;
+import static com.techsophy.tsf.account.constants.ErrorConstants.UN_AUTHORIZED_EXCEPTION;
 
 @RestController
 @AllArgsConstructor(onConstructor_ = {@Autowired})
 public class InternalControllerImpl implements InternalController {
-    private final InternalService internalUserFormDataService;
+    private final UserFormDataServiceImpl userFormDataServiceImpl;
     private final GlobalMessageSource globalMessageSource;
-    private final TokenUtils tokenUtils;
-    private final UserDetails userDetails;
     private final ObjectMapper objectMapper;
     @Override
-    public ApiResponse<InternalUserFormDataSchema> saveUser(String signature, InternalUserFormDataSchema userFormDataSchema, HttpHeaders headers) throws JsonProcessingException {
-        return new ApiResponse<>(internalUserFormDataService.saveUserFormData(signature, userFormDataSchema),true,globalMessageSource.get(SAVE_FORM_SUCCESS));
+    public ApiResponse<UserFormDataSchema> saveUser(String signature, UserFormDataSchema internalUserFormDataSchema, HttpHeaders headers) throws JsonProcessingException
+    {
+        try
+        {
+            Rsa4096 rsa4096 = new Rsa4096();
+            String sign = rsa4096.decryptFromBase64((String) internalUserFormDataSchema.getUserData().get("signature"));
```

Comment:
why signature is coming from user data?

Reply:
When passing Base64 in URL it is converting values such as blank space, and other special characters according to url pattern, and decode is faling, therefore we are passing it in body